### PR TITLE
Fix accepting 1of1 slip39 share

### LINKED ISSUE #455: Support Shamir mnemonics of type 1-of-1
Entering a 1-of-1 mnemonic results in the following screen:
Screenshot from 2019-08-28 14-21-49

Examples:
- "academic academic academic academic academic academic academic academic academic academic academic academic academic academic academic academic academic rebuild aquatic spew"
- "academic academic academic academic academic ocean ocean ocean ocean ocean ocean ocean ocean ocean ocean ocean ocean density email fortune"
- "academic academic academic academic academic hybrid hybrid hybrid hybrid hybrid hybrid hybrid hybrid hybrid hybrid hybrid hybrid ajar hour envy"

### PR DESCRIPTION
**Pull request by ciny**
This PR fixes #455. device tests for this will be added as part of #462

### CODE REVIEW

File: core/src/apps/management/recovery_device/recover.py [hunk 1]
```diff
@@ -44,7 +44,15 @@ def process_slip39(words: str) -> Tuple[Optional[bytes], int, int]:
         )
         storage.recovery_shares.set(index_with_group_offset, words)
 
-        return None, share.group_index, share.index  # we need more shares
+        # if share threshold and group threshold are 1
+        # we can calculate the secret right away
+        if share.threshold == 1 and share.group_threshold == 1:
+            identifier, iteration_exponent, secret, _ = slip39.combine_mnemonics(
+                storage.recovery_shares.fetch()
```
Comment (tsusanka) [inline core/src/apps/management/recovery_device/recover.py]: Any reason to fetch them from storage if we have them in the `words` variable?

File: core/src/apps/management/recovery_device/homescreen.py [hunk 2]
```diff
@@ -191,6 +191,15 @@
...
             await _request_share_next_screen(ctx, is_slip39)
 
+    # if we have a secret and two possible backup_types
+    # it means the slip39 share was 1of1 and the group was 1ofX
+    if len(possible_backup_types) != 1:
+        group_count = storage.recovery.get_slip39_group_count()
+        if group_count == 1:
+            possible_backup_types.remove(BackupType.Slip39_Advanced)
+        else:
+            possible_backup_types.remove(BackupType.Slip39_Basic)
```
Comment (tsusanka) [inline core/src/apps/management/recovery_device/homescreen.py]: Let's keep it like this, I'll modify it during the _request_secret refactor.

File: core/src/apps/management/recovery_device/recover.py (hunk 1 above)
Comment (ciny) [reply core/src/apps/management/recovery_device/recover.py]: fixed.

File: core/src/apps/management/recovery_device/homescreen.py (hunk 2 above)
Comment (ciny) [reply core/src/apps/management/recovery_device/homescreen.py]: ok

**Pull request merged**